xbmc/ServiceBroker.cpp

```diff
@@ -332,3 +332,18 @@ void CServiceBroker::UnregisterCPUInfo()
 {
   g_serviceBroker.m_cpuInfo.reset();
 }
+
+void CServiceBroker::RegisterTextureCache(const std::shared_ptr<CTextureCache>& cache)
+{
+  g_serviceBroker.m_textureCache = cache;
+}
```

Comment:
Should this possibly be 

```
void CServiceBroker::RegisterTextureCache(const std::shared_ptr<CTextureCache> cache)
{
  g_serviceBroker.m_textureCache = std::move(cache);
}
```

I thought i had seen some clang-tidy performance changes in the past and we seem to use that style below in https://github.com/xbmc/xbmc/blob/c041487397533b55bccdbad60a4b0ddc30766df4/xbmc/ServiceBroker.cpp#L326-L329

I'll let others comment on whats the preferred way unless i can find some docs that say one way or the other.

Reply:
There are many discussions in the net around byval+move vs. ref. There is no right and wrong, more or less personal preference. As far as I know not tackled by our coding guidelines. SO, some people use byval+move, others ref. I prefer the latter.